### DISCUSSION TITLE
Plotter write out png or gif on a headless Centos machine

### DISCUSSION THREAD

**Discussion by chen112p** [pyvista/pyvista#1782] [Q&A]
Hello, I understand this topic has been discussed few times. Wondering if anyone made it work on a headless virtual machine with Centos OS. All I am trying to do is to load existing vtk files and put them together as a gif animation.
I have installed Xvfb and mesa-libGL. Also, created a shell script to config Xvfb
![image](https://user-images.githubusercontent.com/39268233/138580257-4a84fb73-8344-47ba-abd4-3c2baa8461ea.png)

But still when running a python script which calls a plotter and write to a png or gif, it tells me 'could not find a decent visual'
![image](https://user-images.githubusercontent.com/39268233/138580302-ea41c459-f690-4dfb-9359-cb8751e77992.png)

I have checked that there is a xvfb process running in the background:
![image](https://user-images.githubusercontent.com/39268233/138580338-53792e81-a516-42d5-b6f7-53853b8c988d.png)

Can you tell what i missed and where i did wrong here?

Thanks!

Junting

**Comment by akaszynski** [comment]
I'm not quite sure why this isn't working, but one alternative would be to build VTK with OSMesa and use that instead.  It doesn't require xvfb.

Building VTK is a bit of a pain, but you can follow:

https://github.com/pyvista/pyvista-wheels

Another approach is to use the pre-built wheel in that repo within a docker image.  See the docker directory.  It works quite well and means that you can use a portable environment and the precompiled wheel.
